app.js

```diff
@@ -53,6 +54,19 @@ app.use(function(req, resp, next){
 
 app.use(reactRouted);
 app.use(express.static(__dirname + '/public', {maxAge: 3600000}));
+app.use(function(req, res, next) {
+  var firstPath = req.url.split("/")[1];
+  var search = req.search || "";
+  // Get a valid locale from the path and header
+  var locale = getLocale(req.headers["accept-language"], firstPath);
+  // Compare the best valid locale to the path,
+  // if they don't match, redirect.
+  if (firstPath !== locale) {
```

Comment:
If these don't match, if means getLocale has decided firstPath is not a locale, and we should redirect to whatever it returned. If they matched, it means getPath decided that the firstPath is already a valid locale.